scripts/pylint_extensions_test.py

```diff
@@ -266,6 +266,111 @@ def test(test_var_one, test_var_two, self): #@
             checker_test_object.checker.visit_functiondef(functiondef_node2)
 
 
+class RestrictedImportCheckerTests(unittest.TestCase):
+
+    def test_detect_restricted_import(self):
+        checker_test_object = testutils.CheckerTestCase()
+        checker_test_object.CHECKER_CLASS = (
+            pylint_extensions.RestrictedImportChecker)
+        checker_test_object.setup_method()
+
+        # Tests the case wherein storage layer imports domain layer
+        # in import statements.
+        node_err_import = astroid.extract_node("""
+            import core.domain.activity_domain #@
+        """)
+        node_err_import.root().name = 'oppia.core.storage.topic'
+        with checker_test_object.assertAddsMessages(
+            testutils.Message(
+                msg_id='invalid-import',
+                node=node_err_import,
+                args=('domain', 'storage'),
+            ),
+        ):
+            checker_test_object.checker.visit_import(node_err_import)
+
+        # Tests the case wherein storage layer does not import domain layer
```

Comment:
Hmm. The test case here does not matches the description: the platform layer is importing domain layer. It should be storage layer _not_ importing domain layer.

Reply:
I am sorry if I didn't get it but isn't it already written? Maybe you are trying to say that the description implies that a storage layer file is not importing domain layer file whereas the test is actually a non-storage layer file importing a domain layer file?
Am I correct? Thanks, making the change!

Reply:
Ah, yes! Thanks!